client/src/extension.ts

```diff
@@ -15,37 +15,32 @@ import { Notifier, Event } from './Notifier';
 
 let fileSystemWatcher: vscode.FileSystemWatcher;
 
-export function activate(context: vscode.ExtensionContext) {
+export function activate(context: vscode.ExtensionContext): Thenable<any> {
+
+	function startServer(): Promise<void> {
+		// create and start Gobra Server
+		fileSystemWatcher = vscode.workspace.createFileSystemWatcher("**/*.{gobra, go}");
+		return State.startLanguageServer(context, fileSystemWatcher);
+	}
+
+	function initVerifier(): Promise<void> {
+		let verifierConfig = new VerifierConfig();
+		Verifier.initialize(context, verifierConfig, fileUri);
+		Notifier.notify(Event.EndExtensionActivation);
+		return Promise.resolve();
+	}
```

Comment:
@Felalolf Yes, I think @ArquintL did this so it can be chained with `.then(initVeriifer)`. However, in JS (and TS as well I would assume?) you can also just chain a regular non-Promise-returning function, so the `return Promise.resolve()` *should* be unnecessary.

Reply:
exactly as Aurel has guessed, I really like the style of `then`-ing a lot of substeps and I create a function for each substep. As this substep indeed only contains synchronous code, I could have written a `void` returning function. I've changed this now